master/docs/manual/customization.rst

```diff
@@ -932,18 +932,72 @@ For more information on the available commands, see :doc:`../developer/master-wo
     Step Progress
     BuildStepFailed
 
-Writing New Status Plugins
---------------------------
+Writing Dashboards with Flask
+-----------------------------
+
+Buildbot Nine UI is written in Javascript.
+This allows it to be reactive and realtime, but comes at a price of a fair complexity.
+Sometimes, you just need a dashboard displaying your build results in your own weird manner but learning angularJS for that is just too much.
+
+There is a buildbot plugin, which allows to write a server side generated dashboard, and integrated it in the UI.
+
+.. code-block:: python
 
-Each status plugin is an object which provides the :class:`twisted.application.service.IService` interface, which creates a tree of Services with the buildmaster at the top [not strictly true].
-The status plugins are all children of an object which implements :class:`buildbot.interfaces.IStatus`, the main status object.
-From this object, the plugin can retrieve anything it wants about current and past builds.
-It can also subscribe to hear about new and upcoming builds.
+    from flask import Flask, render_template
+
+    app = Flask('test', root_path=os.path.dirname(__file__))
+    app.config['TEMPLATES_AUTO_RELOAD'] = True
+    @app.route("/index.html")
+    def main():
+        builders = app.buildbot_api.dataGet("/builders")
+
+        builds = app.buildbot_api.dataGet("/builds", limit=20)
+        # properties are actually not used in the template, but this is how you get more properties
+        for build in builds:
+            build['properties'] = app.buildbot_api.dataGet(("builds", build['buildid'], "properties"))
+
+        return render_template('index.html', builders=builders, builds=builds)
+
+    c['www']['plugins']['flask_dashboard'] = {
+        'dashboards': [
+            {
+                'name': 'mydashboard',  # as used in URLs
```

Comment:
Can I use several dashboards, identified by `name` ..? May be `templates/name.html` for this case..?